typescript-view-swapping/main.ts

```diff
@@ -0,0 +1,30 @@
+const tabContainer = document.querySelector('.tab-container') as HTMLDivElement;
+const tabs = document.querySelectorAll('.tab') as NodeListOf<HTMLDivElement>;
+const views = document.querySelectorAll('.view') as NodeListOf<HTMLDivElement>;
+
+if (!tabContainer) throw new Error('tabContainer query did not work');
+if (!tabs) throw new Error('tabs query did not work');
+if (!views) throw new Error('views query did not work');
+
+tabContainer.addEventListener('click', (event) => {
+  const $eventTarget = event.target as HTMLDivElement;
+
+  if ($eventTarget.matches('.tab')) {
+    tabs.forEach((tab) => {
```

Comment:
This assertion is necessary because we need to access methods on the HTMLDivElement that are not available on just Elements so it is okay.